scripts/internal_links_monitor.py

```diff
@@ -0,0 +1,243 @@
+    def port_counters(self):
+        '''Return the port counters for all the port in the port_map'''
+        port_counters = {}
+        for port_name, port_oid in self.port_map.items():
+            port_counter = self.counter_db.get_all(
+                    "COUNTERS_DB", 'COUNTERS:{}'.format(port_oid))
+            port_counters[port_name] = port_counter
+        return port_counters
+
+    def get_ports_error_above_threshold(self, counter_name, threshold):
+        ''' 
+        Return the list of ports which have the counter value above the threshold
+        '''
+        err_ports = []
+        for port_name, port_counter in self.port_counters.items():
+            try:
+                port_counter_value = port_counter[counter_name]
+                if int(port_counter_value) > int(threshold):
+                    err_ports.append(port_name)
+            except KeyError:
+                print('Bad counter_name')
+        return err_ports
+    
+    def monitor(self, namespace, error_counter_names, threshold ):
+        err_ports = []
+        for counter_name in error_counter_names:
+            err_ports_per_counter = self.get_ports_error_above_threshold(
+                    counter_name,threshold)
+            if err_ports_per_counter:
+                syslog.syslog(syslog.LOG_CRIT,
+                                  ' {} error above threshold on internal port {} in {} '.format(counter_name,err_ports_per_counter, namespace))
+                err_ports.extend(err_ports_per_counter)
+        return err_ports
+
+
+class PacketChassisInternalLinkMontoring():
+    '''A class for monitoring internal links in a Packet Chassis.
+
+    Attributes:
+        link_monitor (dict): A dictionary of LinkMonitor objects, keyed by namespace.
+        error_counter_names (list): A list of error counter names to monitor.
+        threshold (int): The threshold for error counters.
+    '''
+
+    def __init__(self):
+        '''Initializes a PacketChassisInternalLinkMontoring object.
+
+        Args:
+            link_monitor (dict): A dictionary of LinkMonitor objects, keyed by namespace.
+            error_counter_names (list): A list of error counter names to monitor.
+            threshold (int): The threshold for error counters.
+
+        Returns:
+            None
+        '''
+        self.link_monitor = {}
+        self.appdb = {}
+        self.configdb = {}
+        for namespace in self.namespaces:
+            self.appdb[namespace] = self.appl_db(namespace)
+            self.configdb[namespace] = self.config_db(namespace)
+            self.link_monitor[namespace] = InternalLinkErrMontoring(
+                namespace=namespace, port_name_map='COUNTERS_PORT_NAME_MAP',
+                filter_for_backend_ports=self.filter_for_backend_ports)
+        self.error_counter_names = ['SAI_PORT_STAT_IF_IN_ERRORS',
+                                    'SAI_PORT_STAT_IF_OUT_ERRORS']
+        # Using fixed values for 201911
+        self.threshold = 0
```

Comment:
is comment applicable ? Is threshold configurable ?

Reply:
It is fixed for 201911, but would be read from configDB for master and newer branches